From: Rohan Garg <rohan@garg.io>
Date: Tue, 15 Sep 2015 23:20:47 +0000
Subject: Add error checking in various bits so that Baloo doesn't crash when disabled.
X-Git-Url: http://quickgit.kde.org/?p=baloo.git&a=commitdiff&h=3312d6f3f4b80ec04e86c2ea103adc51ade0f020
---
Add error checking in various bits so that Baloo doesn't crash when disabled.

REVIEW: 125241
BUG: 352454
---


--- a/src/engine/database.cpp
+++ b/src/engine/database.cpp
@@ -72,14 +72,24 @@
         return false;
     }
 
-    mdb_env_create(&m_env);
+    int rc = mdb_env_create(&m_env);
+    if (rc) {
+        m_env = 0;
+        return false;
+    }
+
     mdb_env_set_maxdbs(m_env, 12);
     mdb_env_set_mapsize(m_env, static_cast<size_t>(1024) * 1024 * 1024 * 5); // 5 gb
 
     // The directory needs to be created before opening the environment
     QByteArray arr = QFile::encodeName(m_path) + "/index";
-    mdb_env_open(m_env, arr.constData(), MDB_NOSUBDIR | MDB_NOMEMINIT, 0664);
-    int rc = mdb_reader_check(m_env, 0);
+    rc = mdb_env_open(m_env, arr.constData(), MDB_NOSUBDIR | MDB_NOMEMINIT, 0664);
+    if (rc) {
+        m_env = 0;
+        return false;
+    }
+
+    rc = mdb_reader_check(m_env, 0);
     Q_ASSERT_X(rc == 0, "Database::open reader_check", mdb_strerror(rc));
 
     //

--- a/src/lib/searchstore.cpp
+++ b/src/lib/searchstore.cpp
@@ -66,7 +66,7 @@
 
 QStringList SearchStore::exec(const Term& term, int offset, int limit, bool sortResults)
 {
-    if (!m_db) {
+    if (!m_db || !m_db->isOpen()) {
         return QStringList();
     }
 


